fix(docs): Issue 1120 docs guides (#1244)
pmndrs/zustand: docs/guides/typescript.md

)
```

> Also, we recommend using `devtools` middleware as an argument, rather than passing arguments to it. For example, when you use it with `immer` as a middleware, it should be `immer(devtools(...))` and not `devtools(immer(...))`. This is because`devtools` mutates the `setState` and adds a type parameter on it, which could get lost if other middlewares (like `immer`) mutates `setState` before `devtools`.

Reviewer: > we recommend using `devtools` middleware as an argument, rather than passing arguments to it

This sounds a bit wrong, even if you use `devtools` at the end you're still passing an argument to it namely your state creator.

I think the original wording is better, I'm fine if someone can come up with a better one but the one is PR isn't really making the cut imo.

Also "mutate" was correct because it's "middlewares".

```suggestion
Also, we recommend using `devtools` middleware as last as possible. For example, when you use it with `immer`, it should be `immer(devtools(...))` and not `devtools(immer(...))`. This is because `devtools` mutates `setState` and adds a type parameter on it, which could get lost if other middlewares (like `immer`) also mutate `setState` before `devtools`.
```

console.log(useBearStore.foo.toUpperCase())
```

Zustand middlewares can mutate the store. But how could we possibly encode the mutation on the type-level? That is to say how could do we type `foo` so that this code compiles?

> For a usual statically typed language, this is impossible. However, with TypeScript, Zustand has something called a "higher kind mutator" that makes this possible. If you are dealing with complex type problems, like typing a middleware or using the `StateCreator` type, you will have to understand this implementation detail. For this, you can [check out #710](https://github.com/pmndrs/zustand/issues/710).

Reviewer: The original phrasing "thanks to" is better as it implies it's a TypeScript feature more significantly than a mere "with".

And it's [called "higher kinded"](https://www.google.com/search?q=higher+kinded+type) and not "higher kind". Putting a hyphen to make it more correct.

```suggestion
For a usual statically typed language, this is impossible. But thanks to TypeScript, Zustand has something called a "higher-kinded mutator" that makes this possible. If you are dealing with complex type problems, like typing a middleware or using the `StateCreator` type, you will have to understand this implementation detail. For this, you can [check out #710](https://github.com/pmndrs/zustand/issues/710).
```
